src/engines/searx.rs

```diff
@@ -3,16 +3,35 @@
 //! number if provided.
 
 use reqwest::header::HeaderMap;
-use scraper::{Html, Selector};
+use scraper::Html;
 use std::collections::HashMap;
 
+use super::search_result_parser::SearchResultParser;
 use crate::models::aggregation_models::SearchResult;
 use crate::models::engine_models::{EngineError, SearchEngine};
 use error_stack::{Report, Result, ResultExt};
 
 /// A new Searx engine type defined in-order to implement the `SearchEngine` trait which allows to
 /// reduce code duplication as well as allows to create vector of different search engines easily.
-pub struct Searx;
+pub struct Searx {
+    /// The parser, used to interpret the search result.
+    parser: SearchResultParser,
+}
+
+impl Searx {
+    /// creates a Searx parser
```

Comment:
The documentation looks good 👍🏻, but I would like to suggest elaborating it and making it in accordance to the `rust convention` which you can find it here:

 https://rust-lang.github.io/api-guidelines/documentation.html
 
 Also, I would suggest doing it for other functions too 🙂.
 
 Moreover, I would like to recommend taking a look at other files in the project as an example to write documentation following the rust convention. For example, I would suggest taking a look at this file as an example:
 
https://github.com/neon-mmd/websurfx/blob/rolling/src/cache/cacher.rs

Reply:
Gzsombor, are you still working on this suggestion? If not then let us know 🙂 .